TypeSystem/flow-analysis/reachability_try_catch_A03_t02.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion try catch: If `N` is a try/catch statement of the form
+/// `try B alternatives` then:
+/// - Let `before(B) = before(N)`
+/// - For each catch block on `Ti Si` in alternatives:
+///   - Let
+///     `before(Si) = conservativeJoin(before(N), assignedIn(B), capturedIn(B))`
+/// - Let `after(N) = join(after(B), after(C0), ..., after(Ck))`
+///
+/// @description Checks that if a type `T` is made a type of interest in `B`
+/// then it can be promoted in `B`, `alternatives` and `after(N)`.
+/// @author sgrekhov22@gmail.com
+/// @issue 60519
+
+class S {}
+
+class T extends S {
+  int answer() => 42;
+}
+
+class C {
+  T v;
+  C(this.v);
+}
+
+test1() {
+  S s = S();
+  try {
+    if (s is T) {}  // make `T` a type of interest
+    s = T();
+    s.answer();
+  } catch (_) {
+  }
+}
+
+test2() {
+  S s = S();
+  try {
+    if (s is T) {}
+  } catch (_) {
+    (s,) = (T(),);
+    s.answer();
```

Comment:
I would expect this one to fail because `before(S0) = conservativeJoin(before(N), ...)`, where `T` is not a type of interest for `s`.

Reply:
But according to the https://github.com/dart-lang/sdk/issues/60519 this should be an error.

Reply:
Exactly, it should be an error because `T` is not a type of interest for `s` at the location of the pattern assignment.

Reply:
However, I just created https://github.com/dart-lang/language/issues/4328 in order to clarify the situation: The implemented behavior appears to be different from the specified behavior.